homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -48,47 +65,139 @@ async def _async_update_data(self):
             try:
                 weather_response = await self._get_owm_weather()
                 data = self._convert_weather_response(weather_response)
-            except (APICallError, UnauthorizedError) as error:
+            except (APIRequestError, UnauthorizedError) as error:
                 raise UpdateFailed(error) from error
         return data
 
     async def _get_owm_weather(self):
-        weather = await self.hass.async_add_executor_job(
-            self._owm_client.weather_at_coords, self._latitude, self._longitude
-        )
-        return weather.get_weather()
+        """Poll weather data from OWM."""
+        if (
+            self._forecast_mode == FORECAST_MODE_ONECALL_HOURLY
+            or self._forecast_mode == FORECAST_MODE_ONECALL_DAILY
+        ):
+            weather = await self.hass.async_add_executor_job(
+                self._owm_client.one_call, self._latitude, self._longitude
+            )
+        else:
+            current_weather = await self.hass.async_add_executor_job(
+                self._owm_client.weather_at_coords, self._latitude, self._longitude
+            )
+            interval = self._get_forecast_interval()
+            forecast = await self.hass.async_add_executor_job(
+                self._owm_client.forecast_at_coords,
+                self._latitude,
+                self._longitude,
+                interval,
+                self._forecast_limit,
+            )
```

Comment:
Can we avoid two jumps into the executor here?

Reply:
Not with the current version of `pyowm`. This is how the current version of the integration works, and it's essentially because the legacy APIs require two separate API calls, one for current weather, one for forecast.